### PR TITLE
Align heap for GS if necessary only

### PR DESCRIPTION
**Pull request by dmitripivkine**
Usage of GS requires special heap alignments. Usage of Software Barriers
does not have such restrictions as well as calls of
j9gs_enable()/j9gs_disable()

Signed-off-by: Dmitri Pivkine <Dmitri_Pivkine@ca.ibm.com>

### CODE REVIEW

**Comment by dmitripivkine** [comment]
This feature would be activated in OMR https://github.com/eclipse/omr/pull/3067

File: runtime/gc_glue_java/CollectorLanguageInterfaceImpl.cpp [hunk 1]
```diff
@@ -677,45 +677,50 @@ MM_CollectorLanguageInterfaceImpl::scavenger_switchConcurrentForThread(MM_Enviro
 
 	if (_extensions->scavenger->isConcurrentInProgress()) {
 		Assert_MM_true(NULL != _extensions->scavenger);
```
Comment (amicic) [inline runtime/gc_glue_java/CollectorLanguageInterfaceImpl.cpp]: while changing the file, please use _extensions->isConcurrentScavengerInProgress() and remove the assert

**Comment by dmitripivkine** [comment]
Jenkins test sanity zLinux jdk8

**Comment by dmitripivkine** [comment]
Jenkins test sanity zLinux jdk8

**Pull request merged**